### PR TITLE
Save Jobs

### PR DESCRIPTION
**Pull request by marctan13**
Save Jobs, User Auth functions added in store folder(zustand)

Added condition to navbar links to check whether user is seeker/employer to show links

Added mock applications page for employer

Altered login functions in user-store to prevent seeker to log in as employer and vice versa.

formData for post-job and apply pages set up to be sent to back end. (see console log after inputting the forms)

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -55,44 +74,92 @@ function NavBar() {
...
               ml={4} // Distance away from Yapper Logo
             >
-              {colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
+              {colorMode === "light" ? <MoonIcon /> : <SunIcon />}
             </Button>
           </Tooltip>
         </Flex>
         {isLargerThanMobile ? (
           <Flex flex="2" justify="flex-end">
             {/* For Seekers */}
```
Comment (kenoreyes238) [inline src/components/Navbar/Navbar.jsx:89]: don't know if it's just me, but everytime i refresh the page, the navigation links and logout on navbar for both seeker and employer don't show, might be a jwt or fetching problem

File: src/App.jsx [hunk 2]
```diff
@@ -15,8 +16,21 @@ import Applications from "./pages/Applications";
 import ResetPassword from "./pages/ResetPassword";
 import ForgetPassword from "./pages/ForgetPassword";
 import PostJob from "./pages/PostJob";
+import useUserStore from "./store/user-store";
 
 function App() {
+  const { fetchSeeker } = useUserStore(); // Destructure the fetchSeeker function from the user store
+
+  useEffect(() => {
```
Comment (kenoreyes238) [inline src/App.jsx:24]: we probably don't need this useEffect anymore, just comment out in case

File: src/store/user-store.js [hunk 3]
```diff
@@ -1,52 +1,87 @@
 import { create } from "zustand";
 
 const BASE_URL = "http://localhost:3000";
```
Comment (kenoreyes238) [inline src/store/user-store.js:3]: we can probably put the base url in a config or .env file, just a suggestion tho

File: src/App.jsx (hunk 2 above)
Comment (marctan13) [reply src/App.jsx:24]: That would mean the fetch would only run during login.

It wont be able to fetch when you refresh the page for example.

I'll check to see if we need the fetch function when logging in instead

File: src/store/user-store.js (hunk 3 above)
Comment (marctan13) [reply src/store/user-store.js:3]: sounds good. I can make use of the config file you made.

File: src/App.jsx (hunk 2 above)
Comment (kenoreyes238) [reply src/App.jsx:24]: > ah okay gotchu, maybe that's why i wasn't able to refresh the navbar mb

scratch this comment for the useEffect it fixed my navbar 💀

File: src/store/user-store.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+      const data = await response.json();
+      if (!data.jwt) {
+        throw new Error("Failed to log in as seeker");
+      }
+
+      localStorage.setItem("jwt", data.jwt);
+
+      const userData = await get().fetchSeeker(data.jwt);
+      console.log("user data", userData);
```
Comment (kenoreyes238) [inline src/store/user-store.js:32]: we can remove the console.logs for loginSeeker, fetchSeeker, or both, it's running twice

Comment (kenoreyes238) [reply src/store/user-store.js:32]: adn possibly rest too

**Review by kenoreyes238 — APPROVED** [review]

**Pull request merged**